### PR TITLE
Small documentation fixes.

### PR DESCRIPTION
**Pull request by ngnpope**

### CODE REVIEW

**Comment by ngnpope** [comment]
Should be ready now.

**Review by tfranzel — COMMENTED** [review]
woah great work! 

the spec anchors still generate complaints:

```
( drf_spectacular: line    6) broken    https://spec.openapis.org/oas/v3.0.3#parameter-object - Anchor 'parameter-object' not found                                                                                                                                                         
( drf_spectacular: line    8) broken    https://spec.openapis.org/oas/v3.0.3#schema-object - Anchor 'schema-object' not found                                                                                                                                                               
( drf_spectacular: line   11) broken    https://spec.openapis.org/oas/v3.0.3#security-scheme-object - Anchor 'security-scheme-object' not found
```

was it broken before too?

as a sidenote, the changelog shell script generates these links. I simply relied on the github redirects as the numbers are unique identifiers and the script cannot easily distinguish between issues and PRs.

**Comment by ngnpope** [comment]
> was it broken before too?

Yes. In a sense. The issue here is that the anchors on the page are dynamically generated via JavaScript. This means that the linkcheck builder is unable to find them. The original link was pointing to the Markdown file for the spec in GitHub which suffers from the same issue. This is because the anchors for the headings there are stored prefixed with `user-content-`, e.g. `#user-content-parameter-object` which is transformed to `#parameter-object` in JavaScript. You can add the prefix which fixes the output from the linkcheck builder, but it's not that nice. See sphinx-doc/sphinx#9016.

Regardless, I think it's better that we use the documentation hosted under `spec.openapis.org`. Annoyingly, some of the anchors in the raw HTML are kebab-case and some are camelCase, but converted to kebab-case dynamically - and both forms work. I've fixed these to not break the linkcheck.

> as a sidenote, the changelog shell script generates these links. I simply relied on the github redirects as the numbers are unique identifiers and the script cannot easily distinguish between issues and PRs.

Yup. And this is reasonable. I just remembered that `linkcheck_allowed_redirects` is now available. I had wanted it to go further so that you could substitute matches from the source pattern into the target pattern to be more strict, but it'll at least allow us to avoid having to change manually in this case. See sphinx-doc/sphinx#6525 for more details. I've added this setting and reverted some of the redirects to the original. Note that this is also useful for documentation, e.g. Django, where something like `/en/stable/.*` redirects to `/en/4.0/.*` and we always want to point to the latest stable version.

File: docs/conf.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 settings.configure(USE_I18N=False, USE_L10N=False)
 
-sys.path.insert(0, os.path.abspath('../'))
+sys.path[:0] = [os.path.abspath('../'), os.path.abspath('./')]
```
[the remark this answers is not in the document]
Comment (ngnpope) [reply docs/conf.py:20]: It wasn't broken. I added an extension module to handle raising an error when using the default role and it needs to be on the path.

**Comment by tfranzel** [comment]
I see. Thanks for putting in the work!

> Django, where something like /en/stable/.* redirects to /en/4.0/.* and we always want to point to the latest stable version.

yes, noticed that one too.

**Comment by tfranzel** [comment]
for some reason, github did not like the new badge url although request by hand worked without issues. restored old version.

**Comment by ngnpope** [comment]
> for some reason, github did not like the new badge url although request by hand worked without issues. restored old version.

Huh. Weird. Thanks!

**Pull request merged**